### PR TITLE
Update BCEWithLogitsLoss pos_weight description in documentation

### LINKED ISSUE #82496: Required Dimensionality for `pos_weight` conflicts with Documentation on Multi-Dim Labels
### 🐛 Describe the bug

The documentation for the required shape of `pos_weight` in [`BCEwithLogitsLoss`](https://pytorch.org/docs/stable/generated/torch.nn.BCEWithLogitsLoss.html?highlight=bce#torch.nn.BCEWithLogitsLoss) suggests a 1D vector is required:

> pos_weight ([Tensor](https://pytorch.org/docs/stable/tensors.html#torch.Tensor), optional) – a weight of positive examples. Must be a vector with length equal to the number of classes.

In the case where class labels are themselves >1D, however, `pos_weight` requires a tensor with as many dimensions as the labels. I.e. if there are 3 classes of labels that have 2 dimensions each, the shape of `pos_weight` needs to be `([3, None, None])` and a tensor with shape `([3]),` which is the length of the number of classes, will throw an error. (See [this stack overflow question](https://stackoverflow.com/questions/73159568/multi-channel-2d-mask-weights-using-bcewithlogitsloss-in-pytorch/73163289#73163289) for more discussion on this.)

### Versions

PyTorch version: 1.12.0+cu102
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04 LTS (x86_64)
GCC version: (conda-forge gcc 9.4.0-16) 9.4.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.35

Python version: 3.9.13 | packaged by conda-forge | (main, May 27 2022, 16:56:21)  [GCC 10.3.0] (64-bit runtime)
Python platform: Linux-5.4.188-104.359.amzn2.x86_64-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: Tesla T4
Nvidia driver version: 470.57.02
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] torch==1.12.0
[pip3] torch-lr-finder==0.2.1
[pip3] torchsummary==1.5.1
[pip3] torchvision==0.13.0
[conda] cudatoolkit               11.7.0              hd8887f6_10    conda-forge
[conda] efficientnet-pytorch      0.6.3              pyh9f0ad1d_0    conda-forge
[conda] libblas                   3.9.0            15_linux64_mkl    conda-forge
[conda] libcblas                  3.9.0            15_linux64_mkl    conda-forge
[conda] liblapack                 3.9.0            15_linux64_mkl    conda-forge
[conda] magma                     2.5.4                h6103c52_2    conda-forge
[conda] mkl                       2022.1.0           h84fe81f_915    conda-forge
[conda] mkl-service               2.4.0            py39hb699420_0    conda-forge
[conda] numpy                     1.22.1           py39h91f2184_0    conda-forge
[conda] numpydoc                  1.4.0              pyhd8ed1ab_1    conda-forge
[conda] pytorch                   1.10.2          cuda112py39h4de5995_1    conda-forge
[conda] pytorch-gpu               1.10.2          cuda112py39h0bbbad9_1    conda-forge
[conda] pytorch-lightning         1.6.5              pyhd8ed1ab_0    conda-forge
[conda] segmentation-models-pytorch 0.2.1              pyhd8ed1ab_0    conda-forge
[conda] torchgeo                  0.3.0              pyhd8ed1ab_0    conda-forge
[conda] torchmetrics              0.9.3              pyhd8ed1ab_0    conda-forge
[conda] torchvision               0.11.3          cuda112py39ha475c55_2    conda-forge

cc @svekars @holly1238 @albanD @mruberry @jbschlosser @walterddr @kshitij12345 @saketh-are

### PR DESCRIPTION
**Pull request by cviviers**
Fixes #82496 and #65702

cc @svekars @carljparker

### CODE REVIEW

File: torch/nn/modules/loss.py [hunk 1]
```diff
@@ -690,8 +690,10 @@ class BCEWithLogitsLoss(_Loss):
             elements in the output, ``'sum'``: the output will be summed. Note: :attr:`size_average`
             and :attr:`reduce` are in the process of being deprecated, and in the meantime,
             specifying either of those two args will override :attr:`reduction`. Default: ``'mean'``
-        pos_weight (Tensor, optional): a weight of positive examples.
-                Must be a vector with length equal to the number of classes.
+        pos_weight (Tensor, optional): a weight of positive examples. Must be a tensor with equal 
+            dimensions to the target tensor and equal size along the class dimension to the number 
+            of classes. To apply the same positive weight along all spacial dimensions for a 2D
```
Comment (mikaylagawarecki) [inline torch/nn/modules/loss.py:695]: To clarify, must the number of dimensions be equal to that of the target tensor? My understanding is that there are two separate cases here

1) the case mentioned in #65702 where for target of size `[B, C, H, W]` we can either have pos_weight of size `[B, C, H, W]` (where B is batch size) so we apply different pos weights to each element of the batch or `[C, H, W]` (same pos weights across batch)

2) the case mentioned in #82496 where for multidimensional labels `pos_weight` must necessarily have as many dimensions as the label in order to be broadcasted across the spacial dimensions

I think both these cases are in line with the expected [broadcasting semantics](https://pytorch.org/docs/stable/notes/broadcasting.html). Perhaps it makes sense to say that `pos_weight` is broadcasted with target, but draw attention to the broadcasting semantics documentation

Comment (cviviers) [reply torch/nn/modules/loss.py:695]: Ah yes, you are correct. I have updated to PR to reflect this more accurately with the examples. I find the case in https://github.com/pytorch/pytorch/issues/65702 interesting since 'weight' seems to be the intended parameter to manage weighing over batches, although not just for positive examples.

**Review by mikaylagawarecki — COMMENTED** [review]
Thanks! For consistency could you please update the nn.functional docstring [here](https://github.com/pytorch/pytorch/blob/main/torch/nn/functional.py#L3167-L3168) as well 


To fix the failing lint job running `setup_lint` and `quicklint` from your local pytorch directory and following the corresponding instructions should be sufficient, see [here](https://github.com/pytorch/pytorch/blob/4721553431e968ee5e90753346d92252cce98e84/Makefile#L25-L34)

File: torch/nn/modules/loss.py (hunk 1 above)
Comment (mikaylagawarecki) [reply torch/nn/modules/loss.py]: Hm, good catch! Looking at the implementation though, it seems like we first [multiply (and hence broadcast) `pos_weight` with `target`](https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/Loss.cpp#L366-L368) and then [later multiply with weight](https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/Loss.cpp#L374)

**Review by mikaylagawarecki — APPROVED** [review]

**Comment by mikaylagawarecki** [comment]
Hey @cviviers could you run lintrunner to fix the failing lint please!
